src/duk_api_stack.c

```diff
@@ -2339,6 +2340,19 @@ DUK_EXTERNAL const char *duk_to_string(duk_context *ctx, duk_idx_t idx) {
 	return duk_require_string(ctx, idx);
 }
 
+DUK_EXTERNAL const char *duk_to_summary(duk_context *ctx, duk_idx_t idx) {
+	duk_tval *tv;
+
+	DUK_ASSERT_CTX_VALID(ctx);
+
+	idx = duk_require_normalize_index(ctx, idx);
+	tv = DUK_GET_TVAL_POSIDX(ctx, idx);
+	DUK_ASSERT(tv != NULL);
+
+	duk__push_string_tval_readable(ctx, tv, 0 /*error_aware*/);
+	duk_replace(ctx, idx);
```

Comment:
Actually the index *is* reused here.  `duk__push_string_tval_readable()` modifies the stack, then I use `duk_replace()` to overwrite the original value.